bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/LightThingHandler.java

```diff
@@ -0,0 +1,314 @@
+                .thenAccept(v -> logger.trace("Result code={}, body={}", v.getResponseCode(), v.getBody()))
+                .exceptionally(e -> {
+                    logger.debug("Sending command {} to channel {} failed:", command, channelUID, e);
+                    return null;
+                });
+    }
+
+    @Override
+    protected @Nullable LightMessage parseStateResponse(AsyncHttpClient.Result r) {
+        if (r.getResponseCode() == 403) {
+            return null;
+        } else if (r.getResponseCode() == 200) {
+            return gson.fromJson(r.getBody(), LightMessage.class);
+        } else {
+            throw new IllegalStateException("Unknown status code " + r.getResponseCode() + " for full state request");
+        }
+    }
+
+    @Override
+    protected void processStateResponse(@Nullable LightMessage stateResponse) {
+        if (stateResponse == null) {
+            return;
+        }
+        messageReceived(config.id, stateResponse);
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private void valueUpdated(String channelId, LightState newState) {
+        Integer bri = newState.bri;
+        Boolean on = newState.on;
+
+        switch (channelId) {
+            case CHANNEL_SWITCH:
+                if (on != null) {
+                    updateState(channelId, OnOffType.from(on));
+                }
+                break;
+            case CHANNEL_COLOR:
+                Double @Nullable [] xy = newState.xy;
+                Integer hue = newState.hue;
+                Integer sat = newState.sat;
+                if (hue != null && sat != null && bri != null) {
+                    updateState(channelId,
+                            new HSBType(new DecimalType(hue / HUE_FACTOR), toPercentType(sat), toPercentType(bri)));
+                } else if (xy != null && xy.length == 2) {
+                    updateState(channelId, HSBType.fromXY(xy[0].floatValue(), xy[1].floatValue()));
+                }
+                break;
+            case CHANNEL_BRIGHTNESS:
+                if (bri != null && on != null && on) {
+                    updateState(channelId, toPercentType(bri));
+                } else {
+                    updateState(channelId, OnOffType.OFF);
+                }
+                break;
+            case CHANNEL_COLOR_TEMPERATURE:
+                Integer ct = newState.ct;
+                if (ct != null) {
+                    updateState(channelId, new DecimalType(scaleColorTemperature(ct)));
+                }
+                break;
+            case CHANNEL_POSITION:
+                if (bri != null) {
+                    updateState(channelId, toPercentType(bri));
+                }
+            default:
+        }
+    }
+
+    private int scaleColorTemperature(double ct) {
+        return (int) (ct / 100.0 * (500 - 153) + 153);
+    }
+
+    private double scaleColorTemperature(int ct) {
+        return 100.0 * (ct - 153) / (500 - 153);
+    }
```

Comment:
I adjusted the name but I would prefer to keep them non-static. Otherwise one could argue that `toPercentType` and `fromPercentType` should also be static. This would lead to a static logger and I think that is undesirable. Since they are only used within the handler, there is no need to make them static anyway.

Reply:
fine by me